src/main/java/uk/ac/soton/itinnovation/security/modelvalidator/attackpath/AttackPathDataset.java

```diff
@@ -246,65 +221,43 @@ public List<String> getThreatDirectCauseUris(String threatUri) throws RuntimeExc
             ThreatDB threat = threats.get(threatUri);
             return new ArrayList<>(threat.getCausedBy());
         } catch (Exception e) {
-            return new ArrayList<String>();
+            return new ArrayList<>();
         }
     }
 
     /**
-     * check if CSG ends in -Runtime or -Implementation
+     * Check if a Control Strategy Group (CSG) is activated.
      *
-     * @param csgUri
-     * @return
-     */
-    public boolean isCurrentRiskCSG(String csgUri) {
-        return this.checkImplementationRuntime(csgUri);
-    }
-
-    private boolean checkImplementationRuntime(String csgUri) {
-        Pattern pattern = Pattern.compile("\\b-Implementation-Runtime\\b|\\b-Implementation\\b");
-        Matcher matcher = pattern.matcher(csgUri);
-        if (matcher.find()) {
-            return true;
-        } else {
-            return false;
-        }
-    }
-
-    /**
-     * check if CSG ends in -Implementation-Runtime or -Implementation
+     * This method evaluates whether all mandatory Control Sets (CS) associated
+     * with the given CSG are proposed.
      *
-     * @param csgUri
-     * @return
+     * @param csg The Control Strategy Group to be checked.
+     * @return {@code true} if all mandatory Control Sets are proposed,
+     * otherwise {@code false}.
      */
-    public boolean isFutureRiskCSG(String csgUri) {
-        // TODO: REGEX is now changed!!!
-        return !(csgUri.endsWith("-Implementation-Runtime") || csgUri.endsWith("-Implementation"));
+    public boolean isCSGActivated(ControlStrategyDB csg) {
+        return csg.getMandatoryCS().stream().allMatch(cs -> controlSets.get(cs).isProposed());
     }
 
     /**
-     * check if CSG has a contingency plan
+     * Check if control strategy plan exists and is activated need to have a
+     * different way checking for contingency plans
      *
-     * @param csgUri
-     * @return
+     * @param csg the control stragegy
+     * @return {@code true} if contingency plan exists and is activated,
+     * otherwise {@code false}
      */
-    public boolean checkContingencyPlan(String csgUri) throws RuntimeException {
+    public boolean hasContingencyPlan(String csgUri) throws RuntimeException {
```

Comment:
It would be really helpful if a decision such as "we're not going to check if a contingency plan is in place" was recorded along with the reasons on the issue at the time of deciding it.

I don't find the code unreadable, but the code and the Javadoc are not consistent so I do not know which is supposed to be right.